src/lib/yang/alarms.lua

```diff
@@ -348,11 +352,20 @@ end
 -- status change is added to the alarm.
 local function new_alarm (key, args)
    local ret = assert(alarm_list:retrieve(key, args), 'Not supported alarm')
+
+   local alt_resource = args.alt_resource or ret.alt_resource or {}
```

Comment:
According to the YANG schema `alt-resource` is an element of alarm:

```
container alarm-list {
   list alarm {
      key "resource alarm-type-id alarm-type-qualifier";
      uses common-alarm-parameters;
      ...
   }
}
grouping common-alarm-parameters {
   leaf resource {...}
   leaf alarm-type-id {...}
   leaf alarm-type-qualifier {...}
   leaf-list alt-resource {

   }
}
```

The method `alarm_list:retriueve(key, args)` returns an existing alarm, and if doesn't exist it's created. All arguments in `args` are copied to the alarm. So if `declare-alarm` is initialized with an `alt-resource` that list will be copied to an alarm.